### DISCUSSION TITLE
Please Help me with setting up controller/ gamepad

### DISCUSSION THREAD

**Discussion by DefeaterPeter** [LostArtefacts/TRX#803] [Q&A] [answered]
I've tried launching Tomb1Main with my xbox series x controller connected to my windows 10 laptop wireless ( it works fine with minecraft windows game) but Tomb1Main doesn't seem to automatically recognise the controller and i can't get the inputs to work, also same issue when connecting the series x controller wired with usb cable and again this seems to work fine with minecraft windows game, i don't have any idea how to code/ customise tomb1main so if i have to change specific settings to get xbox controller working can somebody please give me an easy to follow walk through guide with steps maybe? To get this working... or would i be best off connecting older xbox 360 or xbox one or ps4, ps5 controller instead? I even have switch pro controller, so plenty of options at home with controllers ( I have no idea what controller is the best to use though for ease/ compatibility). Thanks everyone!

**Comment by walkawayy** [comment, marked as answer]
@DefeaterPeter make sure you are on the latest Tomb1Main https://github.com/LostArtefacts/Tomb1Main/releases. That should allow you to use a controller without any additional setup needed.

Any controllers or gamepads should natively work ~as long as they are connected BEFORE starting the game~ (no longer true, controllers can be connected/disconnected at any time). I don't have an Xbox controller to test, but I know people use them with the game. My PS4/PS5 controllers haven't had an issue (I only use them in wired mode if that makes a difference. Idk how to use Playstation stuff with Bluetooth on Windows). If you're having issues I'm not sure why. When you launch the game, make sure the version number in the bottom right of the main menu is the latest.

**Comment by Pia86** [comment]
I'm having the same problem with my controller (Speedlink Thunderstrike, wired). It works fine with my old TR2 and TR3 installation, but for some reason not in TOMBATI or TR1X or TR2X. The game does not respond at all to any of the buttons, neither arrows nor when I try to customize the controls. 
I downloaded TRX within the last month or so, so I should have the newest version.

**Comment by oziphantom** [comment]
Make sure that is in DInput mode, the old TR2 and TR3 will expect XInput mode, I think that device has a switch on it or a button to change modes. If that doens't work you might need [X360CE](https://www.x360ce.com/) to convert to a more modern input style.

**Comment by Pia86** [comment]
I downloaded Keysticks (freeware). It maps buttons on the controller as though they were the keyboard. So I can use the keyboard controls in TR and yet play with a controller.
